Unable to start receiver expo.modules.taskManager.TaskBroadcastReceiver: java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.android.gms.tasks.Task com.google.android.gms.location.FusedLocationProviderClient.getLastLocation()' on a null object reference

## 🐛 Bug Report

### Summary of Issue <!-- (just a few sentences) -->

Only for some users, the apps crashes randomly on background

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

![image](https://user-images.githubusercontent.com/6757574/99752349-20405680-2b1f-11eb-8044-e1c6e688ff3d.png)

![image](https://user-images.githubusercontent.com/6757574/99752255-f129e500-2b1e-11eb-9232-ba213c925d79.png)

### Reproducible Demo

crashlytics screenshot
![image](https://user-images.githubusercontent.com/6757574/99752060-97c1b600-2b1e-11eb-8ff7-6219265fe39b.png)

sample code
![image](https://user-images.githubusercontent.com/6757574/99753012-6b0e9e00-2b20-11eb-9ed5-c097107a8077.png)

<!--
- This should include as little code as possible, please don't simply link your entire project
- Sharing a link to a [Snack](https://snack.expo.io/) is a GREAT way to provide a reproducible demo :) 
- If a reproducible demo, or a complete list of steps from blank project to bug, are not provided, it is very likely your issue will be closed
- If you need more guidance, please see https://stackoverflow.com/help/mcve
-->
<!--
As an added benefit- creating a repro may help you identify the source of the bug, which means we are one step closer to fixing it! Thanks for helping us help you!
-->
 
### Expected Behavior vs Actual Behavior

The app run normally

I also encountered a similar problem on Android 8.0 (haven't tested on other versions). When the user kill the app and turn off the location service, the app crash immediately. I used EAS build.

```
05-31 23:55:09.669  4932  4932 E AndroidRuntime: Process: <package bundle identifier>, PID: 4932
05-31 23:55:09.669  4932  4932 E AndroidRuntime: java.lang.RuntimeException: Unable to start receiver expo.modules.taskManager.TaskBroadcastReceiver: java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.android.gms.tasks.Task com.google.android.gms.location.FusedLocationProviderClient.getLastLocation()' on a null object reference
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3399)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at android.app.ActivityThread.-wrap18(Unknown Source:0)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1780)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:105)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6944)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.android.gms.tasks.Task com.google.android.gms.location.FusedLocationProviderClient.getLastLocation()' on a null object reference
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at expo.modules.location.taskConsumers.LocationTaskConsumer.didReceiveBroadcast(LocationTaskConsumer.java:114)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at expo.modules.taskManager.TaskService.handleIntent(TaskService.java:313)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at expo.modules.taskManager.TaskBroadcastReceiver.onReceive(TaskBroadcastReceiver.java:15)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3392)
05-31 23:55:09.669  4932  4932 E AndroidRuntime: 	... 8 more
```

expo diagnostics:
```
  Expo CLI 4.5.0 environment info:
    System:
      OS: Linux 5.10 Manjaro Linux
      Shell: 5.1.8 - /bin/bash
    Binaries:
      Node: 14.5.0 - ~/.nvm/versions/node/v14.5.0/bin/node
      Yarn: 1.22.5 - ~/.nvm/versions/node/v14.5.0/bin/yarn
      npm: 7.11.0 - ~/.nvm/versions/node/v14.5.0/bin/npm
    npmPackages:
      expo: ^41.0.0 => 41.0.1 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: https://github.com/expo/react-native/archive/sdk-41.0.0.tar.gz => 0.63.2 
    npmGlobalPackages:
      expo-cli: 4.5.0
    Expo Workflow: managed
```

This code literally do nothing except `console.log`:
```javascript
import * as Location from 'expo-location';
import * as TaskManager from 'expo-task-manager';

const locationTaskName = 'location-tracking';
TaskManager.defineTask(locationTaskName, async (event) => {
    console.log(event.data.locations);
});
```

Is there any update on this bug?

--- a/packages/expo-location/CHANGELOG.md
+++ b/packages/expo-location/CHANGELOG.md
@@ -13,6 +13,7 @@
 ### 🐛 Bug fixes
 
 - Call `jobService.jobFinished` for the finished geofencing jobs. ([#14786](https://github.com/expo/expo/pull/14786) by [@mdmitry01](https://github.com/mdmitry01))
+- Check for null value of `mLocationClient` to prevent a crash ([#15023](https://github.com/expo/expo/pull/15023) by [@zakharchenkoAndrii](https://github.com/zakharchenkoAndrii))
 
 ### 💡 Others
 

--- a/packages/expo-location/android/src/main/java/expo/modules/location/taskConsumers/LocationTaskConsumer.java
+++ b/packages/expo-location/android/src/main/java/expo/modules/location/taskConsumers/LocationTaskConsumer.java
@@ -111,6 +111,10 @@ public class LocationTaskConsumer extends TaskConsumer implements TaskConsumerIn
       maybeReportDeferredLocations();
     } else {
       try {
+        if(mLocationClient == null){
+          Log.w(TAG, "LocationClient is null.");
+          return;
+        }
         mLocationClient.getLastLocation().addOnCompleteListener(new OnCompleteListener<Location>() {
           @Override
           public void onComplete(@NonNull Task<Location> task) {

